packages/jest-environment-webdriver/modules/ConfigureBuilder.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @param {Options} browserOptions
+ * @param {[string, string][]} preferences
+ */
+function setOptionPreferences(browserOptions, preferences) {
+  for (const [key, value] of preferences) {
+    browserOptions.setPreference(key, value);
+  }
+}
+
+function setChromeOptions(builder, options) {
+  const seleniumChrome = require('selenium-webdriver/chrome');
+
+  if (options.capabilities) {
+    builder.withCapabilities(options.capabilities);
+  }
+
+  if (options.driver) {
+    builder.setChromeService(new seleniumChrome.ServiceBuilder(options.driver));
+  }
+
+  const chromeOptions = new seleniumChrome.Options();
+
+  if (options.headless) {
+    chromeOptions.headless();
+  }
+
+  if (options.preferences) {
+    setOptionPreferences(firefoxOptions, options.preferences);
```

Comment:
Shouldn't this be passing in chromeOptions? Same goes for the safari and edge functions.

Reply:
Oh dear, thanks! Updated. Silly copy & paste error on my part.

Reply:
After testing your branch out a bit more I discovered that `setPreference` is Firefox specific. What I actually want to be able to do is provide arguments to the browser via the `addArguments` method [is supported cross browser](https://github.com/SeleniumHQ/selenium/search?l=JavaScript&q=addArguments). I've taken a shot at adding that support based on this PR on my own fork: https://github.com/dmchoull/jest-webdriver/tree/browser-options. I also tried to reduce a bit of the repetition in the browser specific functions. Let me know what you think and feel free to incorporate whatever you want from it into this PR.

Reply:
@JoshuaKGoldberg any thoughts on this? I've been using my fork for a few weeks now but it would be nice to get this functionally incorporated into the main library. The arguments option is essential for me as what I'm testing is a browser extension.